pkg/controller/cell.go

```diff
@@ -7,35 +7,65 @@ import (
 	"github.com/spf13/pflag"
 
 	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
 )
 
 const (
 	controllerGroupMetrics = "controller-group-metrics"
+
+	// labelControllerGroupName is the label used
+	// to identify controller-specific metrics
+	labelControllerGroupName = "group_name"
 )
 
-var ControllerGroupMetrics = map[string]bool{}
+var (
+	// GroupMetricEnabled is populated with the set of ControllerGroups for which metrics are enabled
+	GroupMetricEnabled = map[string]bool{}
+
+	// GroupRuns is a Prometheus-compatible metric for Controller
+	// runs, labeled by completion status and ControllerGroup name
+	GroupRuns metric.Vec[metric.Counter]
+)
 
 var Cell = cell.Module(
 	"controller",
 	"Controllers and Controller Lifecycle management",
 	cell.Config(Config{}),
-	cell.Invoke(ControllerInit),
+	cell.Metric(NewMetrics),
+	cell.Invoke(Init),
 )
 
-// Config contains configuration variables for Controller management
+type Metrics struct {
+	ControllerGroupRuns metric.Vec[metric.Counter]
+}
+
+func NewMetrics() Metrics {
+	return Metrics{
+		ControllerGroupRuns: metric.NewCounterVec(metric.CounterOpts{
+			ConfigName: metrics.Namespace + "_controllers_group_runs_total",
+			Namespace:  metrics.Namespace,
+			Name:       "controllers_group_runs_total",
```

Comment:
How do we know any particular rate() of these is the correct one, since controllers are invoked on different periods and sometimes in exponential backoffs?